app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/MultipleDisposalsTriageController.scala

```diff
@@ -255,6 +256,76 @@ class MultipleDisposalsTriageController @Inject() (
       }
   }
 
+  def countryOfResidence(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    withMultipleDisposalTriageAnswers(request) {
+      case (_, _, answers) =>
+        val wasUkResident = answers.fold(_.wasAUKResident, c => Some(c.countryOfResidence.isUk()))
+
+        if (!wasUkResident.contains(false)) {
+          Redirect(routes.MultipleDisposalsTriageController.wereYouAUKResident())
+        } else {
+          val countryOfResidence =
+            answers.fold(_.countryOfResidence, c => Some(c.countryOfResidence))
+          val form =
+            countryOfResidence.fold(countryOfResidenceForm)(countryOfResidenceForm.fill)
+          Ok(countryOfResidencePage(form, routes.MultipleDisposalsTriageController.wereYouAUKResident()))
+        }
+    }
+  }
+
+  def countryOfResidenceSubmit(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    withMultipleDisposalTriageAnswers(request) {
+      case (_, state, answers) =>
+        countryOfResidenceForm
+          .bindFromRequest()
+          .fold(
+            formWithErrors =>
+              BadRequest(
+                countryOfResidencePage(
+                  formWithErrors,
+                  routes.MultipleDisposalsTriageController.wereYouAUKResident()
+                )
+              ), { countryOfResidence =>
+              if (answers
+                    .fold(_.countryOfResidence, c => Some(c.countryOfResidence))
+                    .contains(countryOfResidence)) {
+                Redirect(routes.MultipleDisposalsTriageController.checkYourAnswers())
+              } else {
+                val updatedAnswers = answers.fold[MultipleDisposalsTriageAnswers](
+                  incomplete =>
+                    incomplete.copy(
+                      countryOfResidence           = Some(countryOfResidence),
+                      wereAllPropertiesResidential = None,
+                      assetType                    = None
```

Comment:
you don't need to unset the `wereAllPropertiesResidential` or the `assetType` here. Same for the complete answers. You should be able to do something like:
```scala
anwers.fold(_.copy(countryOfResidence = ...), _.copy(countryOfResidence = ...))
```